src/BatteryStateOfChargeService.cpp

```diff
@@ -35,29 +39,58 @@ namespace
 BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
 : initialStateOfChargePercent_(initialStateOfChargePercent)
 {
+ AmpHoursUsed_=BATTERY_AMP_HOUR_CAPACITY*(initialStateOfChargePercent_/100);
 }
-
 BatteryStateOfChargeService::~BatteryStateOfChargeService()
 {
 }
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return (AmpHoursUsed_);
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+    if(newCurrent_>=0)
     return false;
+    else
+    return true;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+  double timeLeft;
+  timeLeft=(AmpHoursUsed_/newCurrent_);
+  timeLeft=qAbs(timeLeft);
+  timeLeft=timeLeft/2.77778e-7;
+
+  QTime base(0,0,0);
+  QTime timeOutput= base.addMSecs(timeLeft);
```

Comment:
This new variable `timeOutput` is not really necessary, since `addMSecs` is a member function, using `base.addMSecs` will change base itself. I would suggest replacing this line with:
```c++
return base.addMSecs(timeLeft);
```